src/Mobs/Monster.cpp

```diff
@@ -435,19 +428,41 @@ void cMonster::CalcLeashActions()
 			SetLeashToPos(nullptr);
 		}
 	}
-	else if (IsLeashed())  // Mob is already leashed to an entity: follow it.
+
+	if (!IsLeashed())
+	{
+		return;
+	}
+
+	static const double LeashNaturalLength = 5.0;   // Length at equilibrium
+	static const double LeashMaximumLength = 10.0;  // Length where the leash breaks
+	static const double LeashSpringConstant = 20.0;  // How elastic the leash is
+
+	// If distance to target > Maximum, break leash
+	auto Displacement = m_LeashedTo->GetPosition() - GetPosition();
+	double Distance = Displacement.Length();
+	if (Distance > LeashMaximumLength)
 	{
-		// TODO: leashed mobs in vanilla can move around up to 5 blocks distance from leash origin
-		MoveToPosition(m_LeashedTo->GetPosition());
+		LOGD("Leash broken (distance)");
+		Unleash(false);
+	}
+	else if (Distance > LeashNaturalLength)
+	{
+		// Accelerate monster towards the leashed to entity:
+		auto Extension = Distance - LeashNaturalLength;
+		auto Direction = Displacement / Distance;
+
+		auto Acceleration = Direction * (Extension * LeashSpringConstant);
 
-		// If distance to target > 10 break leash
-		Vector3f a_Distance(m_LeashedTo->GetPosition() - GetPosition());
-		double Distance(a_Distance.Length());
-		if (Distance > 10.0)
+		// Stop mobs from floating up when on the ground
+		if (IsOnGround() && (Acceleration.y < std::abs(GetGravity())))
 		{
-			LOGD("Leash broken (distance)");
-			Unleash(false);
+			Acceleration.y = 0.0;
 		}
+
+		// Apply the acceleration
+		using namespace std::chrono;
```

Comment:
Does this really add enough to justify the extra line?

Reply:
I find it more readable this way.  Cuberite's style conventions don't tend to be economic with line counts anyway.